### ISSUE TITLE
Products downloaded without error but incomplete 

### ISSUE THREAD

**Issue by LattaruoloAndrea** [CDSETool/CDSETool#218]
## Describe the bug
Sometimes the products are partially downloaded and the library doesn't produce any errors. When you check the downloaded products  it is clear that the files are corrupted (See image below).

![image](https://github.com/user-attachments/assets/f95798ad-4ea9-4bc1-a7a2-9f9adbcade85)

In the image the file are downloaded with the correct extension and finished (.SAFE.zip) but the dimensions are too small for sentinel1 products and it is not possible to open them (since the integrity is compromised).

### Suggested fix
It is possible to insert a check on the downloaded part for the checksum or for the dimension of the file downloaded it if is wrong we can use the logic of the retries that we already implemented.

### Versions/ dependencies
- Python 3.11, 
- cdsetool 0.2.13
- windows 11

**Comment by LattaruoloAndrea** [comment]
Maybe this can be useful [219](https://github.com/CDSETool/CDSETool/pull/219)

**Comment by pjonsson** [comment]
Can you reproduce this problem, or does it happen intermittently (and rarely)?

Were the downloads done through `download_features()` so there was concurrency involved?

**Comment by LattaruoloAndrea** [comment]
Yes I used download_features(), so there was concurrency. The problem I would say it happened almost consistently on one of the two PC that I am using for testing.

**Comment by pjonsson** [comment]
My suspicion is that this is caused by errors not propagating properly when concurrency is used (and possibly something Windows-specific). Can you try to reproduce the problem on the PC that often fails but instead use a loop around `download_feature()' getting one product at the time?

**Comment by LattaruoloAndrea** [comment]
I will try and let you know.

**Comment by pjonsson** [comment]
I had a look at the commits of the ThreadPoolExecutor in Python and there seems to be a couple of fixes there that aren't present in Python 3.9. What version of Python is the computer that regularly fails using?

Could you also try replacing `ThreadPoolExecutor` at https://github.com/CDSETool/CDSETool/blob/a6e52e193b0d92cecfe468a97988eab935c9e3d0/src/cdsetool/_processing.py#L34 with `ProcessPoolExecutor` and see if you can reproduce the problem when using concurrency?
